### PR TITLE
'Mirrors now display your reflection' Redux.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
This is a revamp of #62638, which is in turn a remake of #49833, a whooping 3 years old PR, that adds one component to the game, used by mirrors, to do what it reads on the title. Thanks ChungusGamer for reminding me about it.

This time with vampires not being shown in mirrors.

## Why It's Good For The Game
I'm going to post the old screenshots here since I've forgot to take some during the latests tests, while I was busy cussing about a tiny little detail I couldn't achieve (see the documentation on the component path so I don't have to be verbose about it). They look more or less the same, except the visuals look a little less cropped now, ergo nicer-looking:
immagine 
and here another from the previous attempt at PRing this:
Bolshevik Spess Junk LV 2021-11-07 024750

## Changelog

:cl:
image: Mirrors now display a reflection of the mobs next to them... except for vampires.
/:cl:

### CODE REVIEW

File: code/datums/components/reflection.dm [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	SIGNAL_HANDLER
+	QDEL_LIST_ASSOC_VAL(reflected_movables)
+	for(var/atom/movable/target in view(1, source))
+		if(check_can_reflect(target, FALSE))
+			set_reflected(target)
+
+///Checks if the target movable can be reflected or not.
+/datum/component/reflection/proc/check_can_reflect(atom/movable/target, check_view = TRUE)
+	var/atom/movable/mov_parent = parent
+	if(target == src || (check_view && !(target in view(1, parent))))
```
Comment (MrMelbert) [inline code/datums/components/reflection.dm:86]: `target == src`, you probably want `target == parent`

**Comment by necromanceranne** [comment]
We're in the big leagues of game development now.

**Review by Wallemations — APPROVED** [review]

**Comment by casualspacestation14enjoyer** [comment]
cant wait for shitsec forcing everyone to go next to a mirror to check for vampires

**Comment by LSommers** [comment]
Imagine if they had a low chance to reflect ghosts.
I think I'd about shit myself if I saw that unexpectedly.

**Comment by DrDiasyl** [comment]
> cant wait for shitsec forcing everyone to go next to a mirror to check for vampires

First get vampires.

**Pull request merged**